### DISCUSSION TITLE
Nix and Framework-X

### DISCUSSION THREAD

**Discussion by drupol** [clue/framework-x#132] [Show and tell]
Hi,

I've been playing with Nix since a couple of months and I had the idea to test how to package a PHP application.

[Nix](https://nixos.org/nix/) consists of a package manager and a language to describe packages for the said manager. It features reproducible builds, cross-distro and -platform compatibility, binary caching, as well as a large collection of packages maintained by thousands of contributors.

Nix can be installed on any Linux, Windows and MacOS. 

[Flakes](https://www.tweag.io/blog/2020-05-25-flakes/) is the upcoming features of Nix, and I'm currently playing with it since it's very stable.

If you want to give it a try with clue/framework-x, you just to do: `nix run github:drupol/framework-x/feat/use-nix`

This will run the `clue/framework-x` demo locally and it will be available at http://127.0.0.1:8080/

If you're curious, you can check the things I had to do in order to get this working in this PR: https://github.com/drupol/framework-x/pull/1

## Docker

To build a docker image containing everything to run the application, do:

```
nix build github:drupol/framework-x/feat/use-nix#oci -o oci-image.tar.gz
docker load -i oci-image.tar.gz
docker run -p 8080:8080 clue/framework-x:20220408152914-009cc71
```

PS: I also did the same for the symfony/demo: `nix run github:drupol/demo/feat/use-nix`

Let me know about what you think!

**Comment by SimonFrings** [comment]
@drupol Thanks for sharing :+1: 

Looks great, the more options the better! Maybe this is also something for the future of Framework X.
I'm curious what others have to say about this!

**Reply by drupol** [reply]
I just added the steps to build a full featured layered docker image by just running one single command.
